tasks/sync-granule/GranuleFetcher.js

```diff
@@ -36,47 +29,61 @@ const addChecksumToFile = async (providerClient, checksumFiles, dataFile) => {
 };
 
 const addChecksumsToFiles = async (providerClient, files) => {
-  const checksumFiles = files.filter(isChecksumFile);
-  const dataFiles = files.filter((f) => !isChecksumFile(f));
-
-  const updatedDataFiles = await Promise.all(
-    dataFiles.map((dataFile) =>
-      addChecksumToFile(providerClient, checksumFiles, dataFile))
+  // Map data file name to checksum file object, if it has a checksum file
+  const checksumFileOf = files.reduce((acc, file) => {
+    if (isChecksumFile(file)) {
+      const ext = path.extname(file.name);
+      const dataFilename = path.basename(file.name, ext);
+      acc[dataFilename] = file;
+    }
+    return acc;
+  }, {});
+
+  return Promise.all(
+    // Map over all files in order to preserve original file ordering
+    files.map((file) => addChecksumToFile(
+      providerClient, checksumFileOf[file.name], file
+    ))
   );
-
-  return [...checksumFiles, ...updatedDataFiles];
 };
 
 class GranuleFetcher {
   /**
-   * Constructor for GranuleFetcher class
+   * Constructor for GranuleFetcher class.
    *
-   * @param {Object} buckets - s3 buckets available from config
-   * @param {Object} collection - collection configuration object
-   * @param {Object} provider - provider configuration object
-   * @param {string} fileStagingDir - staging directory on bucket,
-   * files will be placed in collectionId subdirectory
-   * @param {boolean} duplicateHandling - duplicateHandling of a file
+   * @param {Object} options - keyword options argument
+   * @param {Object} options.buckets - s3 buckets available from config
+   * @param {Object} options.collection - collection configuration object
+   * @param {Object} options.provider - provider configuration object
+   * @param {string} options.fileStagingDir - staging directory on bucket,
+   *    files will be placed in collectionId subdirectory
+   * @param {string} options.duplicateHandling - duplicateHandling of a file;
+   *    one of 'replace', 'version', 'skip', or 'error' (default)
+   * @param {boolean} options.syncChecksumFiles - if `true`, also synchronize
+   *    checksum files (default: `false`)
    */
-  constructor(
+  constructor({
     buckets,
     collection,
     provider,
     fileStagingDir = 'file-staging',
-    duplicateHandling = 'error'
-  ) {
+    duplicateHandling = 'error',
+    syncChecksumFiles = false
```

Comment:
Given that we only use `syncChecksumFiles` in the `ingest` method, consider making it a parameter to that method rather than making it a property of the `GranuleFetcher` object.